modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/resource/releases/SW360Release.java

```diff
@@ -201,30 +210,77 @@ public void setCopyrights(String copyrights) {
     }
 
     public boolean shareIdentifier(SW360Release releaseCompare) {
-        return this.componentId.equals(releaseCompare.getComponentId())
-                && this.name.equals(releaseCompare.getName())
-                && this.version.equals(releaseCompare.getVersion());
+        return this.name.equals(Optional.of(releaseCompare.getName()).orElse(""))
+                && this.version.equals(Optional.of(releaseCompare.getVersion()).orElse(""));
+    }
+
+    public SW360Release mapExternalIdsOnVariable() {
+        externalIds.forEach((key, value) -> {
+            switch (key) {
+                case SW360Attributes.RELEASE_EXTERNAL_ID_FLICENSES:
+                    finalLicense = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_DLICENSES:
+                    declaredLicense = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_OLICENSES:
+                    observedLicense = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_OREPO:
+                    releaseTagUrl = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_SWHID:
+                    softwareHeritageId = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_CHANGESTATUS:
+                    changeStatus = value;
+                    break;
+                case SW360Attributes.RELEASE_EXTERNAL_ID_COPYRIGHTS:
+                    copyrights = value;
+                    break;
+                default:
+                    if (key.startsWith(SW360Attributes.RELEASE_EXTERNAL_ID_HASHES)) {
+                        if(hashes == null) {
+                            hashes = Collections.singleton(value);
+                        } else {
+                            hashes.add(value);
+                        }
+                    } else if (SW360CoordinateKeysToArtifactCoordinates.getValues().contains(key)) {
+                        if(coordinates == null) {
+                            coordinates = new HashMap<String, String>();
+                            coordinates.put(key, value);
+                        } else {
+                            coordinates.put(key, value);
+                        }
+                        coordinates = Collections.unmodifiableMap(coordinates);
```

Comment:
Here you make the coordinates immutable, so the second time one wants to put something (e.g. in line 253) it will fail